### ISSUE TITLE
State out of sync when deleting from SetSchema 

### ISSUE THREAD

**Issue by thomasjonas** [colyseus/colyseus#407]
We've noticed a bug with SetSchema where the state between server and client gets out of sync. Initially we thought it was only happening when you would both add and delete from a SetSchema, but it seems more related to deleting. Although it does feel a bit random. When I run this code on the server ([or more precise](https://github.com/thomasjonas/colyseus-examples/commit/7e6d4971356a9f06ef923308ffb63440ff42346b)):

```
this.players.get(sessionId).skills.delete("A");
this.players.get(sessionId).skills.add("D");
this.players.get(sessionId).skills.delete("B");
this.players.get(sessionId).skills.add("E");
this.players.get(sessionId).skills.delete("C");
this.players.get(sessionId).skills.add("F");
console.log(this.players.get(sessionId).skills.toArray());
```

The server has `[ 'D', 'E', 'F' ]` as state, while the client has `[ 'A', 'B', 'C', 'D', 'E', 'F' ]` as state. I've forked the examples repo that should reproduce the bug.

## Steps to Reproduce (for bugs)
1. Fork my repository at https://github.com/thomasjonas/colyseus-examples
2. run `npm install` and `npm start`
3. Open the state handler example (http://localhost:2567/02-state-handler.html)
4. In the bottom right corner you should see the current "skills set" `A,B,C`
5. Make 6 or more moves
6. The skill set on the client now shows `A,B,C,D,E,F` while the logs on the server say `[ 'D', 'E', 'F' ]`. The server is correct, so somehow the client is not in sync.

## Your Environment
* Colyseus Version: colyseus-0.14.13
* Node.js Version: v15.8.0
* Operating System and version: macOS Catalina (10.15.7)

**Comment by endel** [comment]
Thanks a lot for reporting and providing a reproducible example @thomasjonas! I've just released `@colyseus/schema@1.0.22` fixing this issue!

After using the latest version you should see the deletions you've prepared from the client-side:

<img width="309" alt="Screen Shot 2021-04-14 at 00 17 14" src="https://user-images.githubusercontent.com/130494/114649382-d1a27d00-9cb6-11eb-87ef-411785f36fb5.png">

Please let me know if you happen to encounter any other issues! Cheers!

**Comment by thomasjonas** [comment]
Great! Thank you so much!
